File: android-navigator-compiler/src/main/java/com/github/kostasdrakonakis/compiler/IntentPropertyData.java
package com.github.kostasdrakonakis.compiler;

class IntentPropertyData {
    private String fieldName;
    private String annotationValue;
    private String fieldClass;

    private boolean hasIntValue;
    private boolean hasDoubleValue;
    private boolean hasFloatValue;
    private boolean hasLongValue;
    private boolean hasCharValue;
    private boolean hasShortValue;
    private boolean hasByteValue;
    private boolean hasBooleanValue;

    private double doubleDefaultValue;
    private float floatDefaultValue;
    private long longDefaultValue;
    private char charDefaultValue;
    private short shortDefaultValue;
    private byte byteDefaultValue;
    private boolean booleanDefaultValue;
    private int intDefaultValue;

    IntentPropertyData(String fieldName, String annotationValue, String fieldClass) {
        this.fieldName = fieldName;
        this.annotationValue = annotationValue;
        this.fieldClass = fieldClass;
    }

    String getFieldName() {
        return fieldName;
    }

    String getAnnotationValue() {
        return annotationValue;
    }

    String getFieldClass() {
        return fieldClass;
    }

    public int getIntDefaultValue() {
        return hasIntValue ? intDefaultValue : 0;
    }

    public double getDoubleDefaultValue() {
        return hasDoubleValue ? doubleDefaultValue : 0.0d;
    }

    public float getFloatDefaultValue() {
        return hasFloatValue ? floatDefaultValue : 0.0f;
    }

    public long getLongDefaultValue() {
        return hasLongValue ? longDefaultValue : 0L;
    }

    public char getCharDefaultValue() {
        return hasCharValue ? charDefaultValue : '\u0000';
    }

    public short getShortDefaultValue() {
        return hasShortValue ? shortDefaultValue : 0;
    }

    public byte getByteDefaultValue() {
        return hasByteValue ? byteDefaultValue : 0;
    }

    public boolean getBooleanDefaultValue() {
        return hasBooleanValue && booleanDefaultValue;
    }

    public void setIntDefaultValue(int intDefaultValue) {
        hasIntValue = true;
        this.intDefaultValue = intDefaultValue;
    }

    public void setDoubleDefaultValue(double doubleDefaultValue) {
        hasDoubleValue = true;
        this.doubleDefaultValue = doubleDefaultValue;
    }

    public void setFloatDefaultValue(float floatDefaultValue) {
        hasFloatValue = true;
        this.floatDefaultValue = floatDefaultValue;
    }

    public void setLongDefaultValue(long longDefaultValue) {
        hasLongValue = true;
        this.longDefaultValue = longDefaultValue;
    }

    public void setCharDefaultValue(char charDefaultValue) {
        hasCharValue = true;
        this.charDefaultValue = charDefaultValue;
    }

    public void setShortDefaultValue(short shortDefaultValue) {
        hasShortValue = true;
        this.shortDefaultValue = shortDefaultValue;
    }

    public void setByteDefaultValue(byte byteDefaultValue) {
        hasByteValue = true;
        this.byteDefaultValue = byteDefaultValue;
    }

    public void setBooleanDefaultValue(boolean booleanDefaultValue) {
        hasBooleanValue = true;
        this.booleanDefaultValue = booleanDefaultValue;
    }
}

File: android-navigator-compiler/src/main/java/com/github/kostasdrakonakis/compiler/IntentPropertyData.kt
package com.github.kostasdrakonakis.compiler

internal class IntentPropertyData(val fieldName: String, val annotationValue: String, val fieldClass: String) {
    private var hasIntValue = false
    private var hasDoubleValue = false
    private var hasFloatValue = false
    private var hasLongValue = false
    private var hasCharValue = false
    private var hasShortValue = false
    private var hasByteValue = false
    private var hasBooleanValue = false
    var doubleDefaultValue = 0.0
        get() = if (hasDoubleValue) field else 0.0
        set(doubleDefaultValue) {
            hasDoubleValue = true
            field = doubleDefaultValue
        }
    var floatDefaultValue = 0f
        get() = if (hasFloatValue) field else 0.0f
        set(floatDefaultValue) {
            hasFloatValue = true
            field = floatDefaultValue
        }
    var longDefaultValue: Long = 0
        get() = if (hasLongValue) field else 0L
        set(longDefaultValue) {
            hasLongValue = true
            field = longDefaultValue
        }
    var charDefaultValue = 0.toChar()
        get() = if (hasCharValue) field else '\u0000'
        set(charDefaultValue) {
            hasCharValue = true
            field = charDefaultValue
        }
    var shortDefaultValue: Short = 0
        get() = if (hasShortValue) field else 0
        set(shortDefaultValue) {
            hasShortValue = true
            field = shortDefaultValue
        }
    var byteDefaultValue: Byte = 0
        get() = if (hasByteValue) field else 0
        set(byteDefaultValue) {
            hasByteValue = true
            field = byteDefaultValue
        }
    var booleanDefaultValue = false
        get() = hasBooleanValue && field
        set(booleanDefaultValue) {
            hasBooleanValue = true
            field = booleanDefaultValue
        }
    var intDefaultValue = 0
        get() = if (hasIntValue) field else 0
        set(intDefaultValue) {
            hasIntValue = true
            field = intDefaultValue
        }

}